docs/mds_test_plan.md

```diff
@@ -0,0 +1,39 @@
+# Metadata Service (mds) test plan
+
+### One-liner overview of the service
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [[heading-increment] Heading levels should increment by one level at a time](https://app.codacy.com/gh/uc-cdis/metadata-service/pullRequest?prid=5189388)